src/main/java/com/appium/manager/TestLogger.java

```diff
@@ -207,18 +207,22 @@ private void handleTestFailure(ITestResult result, String className,
                                 .getMethodName() + "/" + screenShotNameWithTimeStamp + deviceModel
                                 + "_failed_" + result.getMethod().getMethodName() + "_framed.png");
                 if (framedImageAndroid.exists()) {
-                    log.addScreenCaptureFromPath(
-                            "screenshot/android/" + device_udid.replaceAll("\\W", "_") + "/"
+                    test.get().log(Status.FAIL,"Snapshot below: " 
+                        + ExtentTestManager.getTest().addScreenCaptureFromPath(
+                            System.getProperty("user.dir") + "/target/screenshot/android/" 
+                            + device_udid.replaceAll("\\W", "_") + "/"
                                     + className + "/" + result.getMethod().getMethodName()
                                     + "/" + screenShotNameWithTimeStamp
                                     + deviceModel + "_failed_" + result
-                                    .getMethod().getMethodName() + "_framed.png");
+                                    .getMethod().getMethodName() + "_framed.png"));
                 } else {
-                    log.addScreenCaptureFromPath(
-                            "screenshot/android/" + device_udid.replaceAll("\\W", "_") + "/"
+                    test.get().log(Status.FAIL,"Snapshot below: " 
+                        + ExtentTestManager.getTest().addScreenCaptureFromPath(
+                            System.getProperty("user.dir") + "/target/screenshot/android/" 
+                            + device_udid.replaceAll("\\W", "_") + "/"
```

Comment:
Why are you keeping all this repeated code and logic? This does need a refactor.